### DISCUSSION TITLE
How to invoke double/int64?

### DISCUSSION THREAD

**Discussion by 0xzer** [vfsfitvnm/frida-il2cpp-bridge#326] [Q&A]
Is it possible to invoke a class constructor with set `System.Double` and `System.Int64` values? I tried `Il2cpp.Reference.to(..., ...)` but I had no success in being able to pass those values to invoke because it returned `Error: Expected number`. I tried both `.value` and just the reference itself.

**Comment by vfsfitvnm** [comment]
Just pass the numbers as they are

**Reply by 0xzer** [reply]
this doesn't work, maybe it's because it's a `Struct`? I can access it using `.class` though (cus C# is object oriented i presume?)
after implementing a function in the struct, the field values are just 0 0 they are never set to what I set them to

**Reply by vfsfitvnm** [reply]
Hey I have no context here. What are you trying to do? If you want to create an object (a class instance) and you need to pass arguments to any of its constructors:
```ts
const Class = ...;
const instance: Il2Cpp.Object = Class.alloc();
instance.method(".ctor").invoke(12345);

// if 'Class.isValueType == true', remember to pass 'instance.unbox()' (instead of 'instance') when passing the instance as parameter to another method!
```
